src/nl/rubensten/texifyidea/action/ShowEquationPreview.kt

```diff
@@ -0,0 +1,112 @@
+package nl.rubensten.texifyidea.action
+
+import com.intellij.lang.folding.FoldingDescriptor
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.SelectionModel
+import com.intellij.openapi.fileEditor.TextEditor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.wm.ToolWindow
+import com.intellij.openapi.wm.ToolWindowAnchor
+import com.intellij.openapi.wm.ToolWindowManager
+import com.intellij.psi.PsiElement
+import com.intellij.ui.content.ContentFactory
+import nl.rubensten.texifyidea.TexifyIcons
+import nl.rubensten.texifyidea.psi.LatexMathEnvironment
+import nl.rubensten.texifyidea.psi.LatexPsiUtil
+import nl.rubensten.texifyidea.window.EquationPreviewToolWindow
+import nl.rubensten.texifyidea.window.PreviewForm
+import org.jetbrains.jsonProtocol.prepareWriteRaw
+import org.picocontainer.Disposable
+
+
+/**
+ * @author Ruben Schellekens
+ */
+class ShowEquationPreview : EditorAction("Equation preview", TexifyIcons.EQUATION_PREVIEW) {
+
+    companion object {
+        @JvmStatic
+        lateinit var FORM_KEY: Key<PreviewForm>
+    }
+
+    init {
+        FORM_KEY = Key<PreviewForm>("form")
```

Comment:
```suggestion
        FORM_KEY = Key("form")
```
Type not necessary.

Reply:
True. But type is required to fetch the value. I choose to embed type information in the key. In my taste latter is a bit better. 

Of-topic: Eventually I had troubles with `content.putUserData(Key("form"),value)` and `val value = content.getUserData<ValueType>(Key("form"))`. The `getUserData` always returned `null` that way. Seems like to fetch the value one must provide exactly same instance of the `Key`. Is it by design?

Reply:
You are right, fixed.